test/export/test_export.py

```diff
@@ -1721,6 +1722,42 @@ def forward(self, scores, mask):
         optimized_model = torch.compile(exported_model)
         optimized_model(tensor_cpu, mask_cpu)
 
+    def test_export_input_mutation_static_shape(self):
```

Comment:
This might be too edge case-y: What happens when you have a user input that doesn't require grad but later mutated to require grad? I remember something about buffer is not allowed to have a grad or sth?

Reply:
I just tested with this case and it seems to fail on dynamo even before aot autograd: "a view of a leaf Variable that requires grad is being used in an in-place operation."